grpc4bmi/bmi_julia_model.py

```diff
@@ -0,0 +1,669 @@
+from typing import List
+
+from bmipy import Bmi
+import numpy as np
+from juliacall import Main as jl, ModuleValue, TypeValue
+
+class BmiJulia(Bmi):
+    """Python Wrapper of a Julia based implementation of BasicModelInterface.
+
+    BasicModelInterface is available in https://github.com/Deltares/BasicModelInterface.jl repo.
+
+    Args:
+        model: Julia model class
+        implementation: Julia variable which implements BasicModelInterface
+    """
+    state = None
+
+    @classmethod
+    def from_name(cls, model_name, implementation_name = 'BasicModelInterface'):
+        """Construct BmiJulia from Julia model class name and implementation name.
+
+        Args:
+            model_name: Name of Julia model class
+                The package of model_name should be installed.
+            implementation_name: Name of Julia variable which implements BasicModelInterface
+                The package of implementation_name should be installed.
+                Uses https://juliapackages.com/p/basicmodelinterface by default.
+        """
+        package4model = model_name.split('.')[0]
+        package4implementation = implementation_name.split('.')[0]
+        jl.seval("import " + package4model)
+        jl.seval("import " + package4implementation)
+        model = jl.seval(model_name)
+        implementation = jl.seval(implementation_name)
+        return BmiJulia(model, implementation)
```

Comment:
One problem with this: you cannot add this to an eWaterCycle LocalModel. That requires a callable class, i.e. one that is not initialized yet.

For Wflow.jl I had to do the following to make it work with eWaterCycle:
```py
class WflowBmi(BmiJulia):
    def __init__(self):
        m = self.from_name("Wflow.Model", implementation_name="Wflow.BMI")
        super().__init__(m.model, m.implementation)
```

Reply:
ah, it made sense for grpc server from cli, but not as a super class.

Created https://github.com/eWaterCycle/grpc4bmi/issues/145